pkg/storage/storage_get.go

```diff
@@ -21,11 +21,14 @@ type GetInput struct {
 	EndTime   time.Time
 	Key       *segment.Key
 	Query     *flameql.Query
+	// TODO: make this a part of the query
+	GroupBy string
```

Comment:
Maybe we should make this a slice, like in the most of query languages. I anticipate that many users will find this useful. Another very handy operator is `without`, with makes query engine to ignore the specified labels. I'm not sure if we need this (and, especially, in the first version), just a noting.

Also, I wonder if we're still going to maintain _similarity_ to PromQL. I think that it may make sense to go our own way. Otherwise, we should follow [their semantics](https://prometheus.io/docs/prometheus/latest/querying/operators/#aggregation-operators) (e.g there `group by` is an aggregation operator and does slightly different things).

Reply:
Good points. Will make it a slice in the future, it should be backward-compatible with the HTTP API (e.g `groupBy=foo,bar`. Same with `without`, (e.g `groupBy=!foo`).

Eventually we might add it to flameQL too. I think it's too early for these decisions tho. At this point my goal with this current design is to make it so that we *can* add all these additional features in the future.